sw/device/lib/runtime/epmp.h

```diff
@@ -0,0 +1,458 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_RUNTIME_EPMP_H_
+#define OPENTITAN_SW_DEVICE_LIB_RUNTIME_EPMP_H_
+
+#include <stdint.h>
+
+/**
+ * RISC-V Enhanced Physical Memory Protection (EPMP).
+ *
+ * Specifications:
+ *  - PMP Enhancements for memory access and execution prevention on Machine
+ *    mode
+ *    https://docs.google.com/document/d/1Mh_aiHYxemL0umN3GTTw8vsbmzHZ_nxZXgjgOUzbvc8
+ *  - RISC-V Privileged Specfication (v. 20190608)
+ *    https://github.com/riscv/riscv-isa-manual/releases/download/Ratified-IMFDQC-and-Priv-v1.11/riscv-privileged-20190608.pdf
+ *  - Ibex PMP
+ *    https://ibex-core.readthedocs.io/en/latest/03_reference/pmp.html
+ *
+ * Assumptions (should be initialized in assembly but can be verified using
+ * `epmp_check`):
+ *   * Machine Mode Whitelist Policy is enabled (mseccfg.MMWP = 1)
+ *   * Machine Mode Lockdown is disabled (mseccfg.MML = 0)
```

Comment:
In that case do you think we can leave the `RLB` toggle out of the API or do you think we'll need to set `RLB = 0` to lock the rules at some point?

Related: should we just assume all entries configured using this API are locked (i.e. `L = 1`)?

Reply:
> Related: should we just assume all entries configured using this API are locked (i.e. L = 1)?

I think this is reasonable.

> In that case do you think we can leave the RLB toggle out of the API

I am not entirely sure at which stage of Secure Boot we will clear it, so I think it is worth keeping for now. We can always revisit it later.

Reply:
>  I think we should assume that we start with RLB = 1 as well as MMWP = 1.

I agree.

Also, `MML` is sticky and `0` on reset, right? Maybe we should just say "MML is not set" because the comment sounds like assembly code must set it to `0`.

> In that case do you think we can leave the RLB toggle out of the API or do you think we'll need to set RLB = 0 to lock the rules at some point?

IMHO we should have it in the API but as a disable-only function just to make the intent clearer (`RLB` is locked when it is cleared if there is a rule with the `L` bit set anyway but I don't think we have a use-case for setting it again, do we?)